pytest_mpi/__init__.py

```diff
@@ -1 +1,4 @@
 from .parallel_assert import parallel_assert  # noqa: F401
+from .detect_mpiexec_implementation import detect_mpiexec_implementation
+
+impl = detect_mpiexec_implementation()
```

Comment:
I think we should avoid doing this at import time. If someone doesn't have MPI installed but they do have this plugin (unlikely I know) then this could break their system. Also some systems don't call `mpiexec` and use `srun` instead. I think a lazy global variable inside `plugin.py` would be better:
```py
_mpi_implementation = None

def _set_parallel_callback(...):
    global _mpi_implementation

    if _mpi_implementation is None:
        _mpi_implementation = sniff_mpi_implementation()

    ...
```